# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

<a id = 'mean'></a>

Рассчитаем медианное значение трудового стажа по типу занятости.

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

Рассчитаем среднеарифметическое значение трудового стажа по типу занятости.

In [8]:
data.groupby('income_type')['days_employed'].describe()

count           mean           std            min  \
income_type                                                            
безработный          2.0  366413.652744  40855.478519  337524.466835   
в декрете            1.0    3296.759962           NaN    3296.759962   
госслужащий       1312.0    3399.896902   2788.371363      39.954170   
компаньон         4577.0    2111.524398   2048.448594      30.195337   
пенсионер         3443.0  365003.491245  21069.606065  328728.720605   
предприниматель      1.0     520.848083           NaN     520.848083   
сотрудник        10014.0    2326.499216   2307.924129      24.141633   
студент              1.0     578.751554           NaN     578.751554   

                           25%            50%            75%            max  
income_type                                                                  
безработный      351969.059790  366413.652744  380858.245699  395302.838654  
в декрете          3296.759962    3296.759962    3296.759962    3296.759962  
госслужащий        1257.171811    2689.368353    4759.399260   15193.032201  
компаньон           685.687432    1547.382223    2876.648520   17615.563266  
пенсионер        346649.346146  365213.306266  383231.396871  401755.400475  
предприниматель     520.848083     520.848083     520.848083     520.848083  
сотрудник           746.027361    1574.202821    3108.123025   18388.949901  
студент             578.751554     578.751554     578.751554     578.751554

У двух типов (безработные и пенсионеры) среднеарифметическое также как и у медианы получатся аномально большим.

По трем группам (госслужащие, компаньены и сотрудники) среднеарифметическое и медиана не соврадают. Во всех случаях не совпадения медианное значение меньше чем среднеарифметическое. Миксимальные значения по этим группам сильно завышены, значит в выборке есть выбросы, они искажают сренеарифметическое.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [9]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [10]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [11]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [12]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [13]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [14]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [15]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [16]:
data.duplicated().sum()

71

In [17]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [18]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [19]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [20]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [21]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [22]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

Задача прогнозирования надежности заемщика представляет собой задачу классификации.

Для того, чтобы от сортировать клиентов по риску невозврата (или просрочек) кредита, составим персональный рейтинг клиентов сформировав группы по столбцам `семейное положение`, `цель кредита`, `количество детей в семье`, определим по группам количество просроченных кредитов, найдем процент клиентов от текущей группы, которые имеют задолженность.

Все категориальные данные рассматриваемых столбцов преобразуем в числовое значения, где каждому классу будет соответствовать единственное число - балл(рейтинг). Таким образом предложим банку свою балльную систему (рейтингов) на основании выдвинутых и подтвержденных гипотез.
 
Выдвинем гипотезы, докажем или опровергнем их.

Для исследования гипотез, будем использоваться  одни и те же шаги для анализа данных, поэтому объявим метод, который сформирует таблицу с данными для анализа. 
Метод будет принимать на вход датафрейм и столбец для анализа, результатом использования метода будет таблица с данными для анализа:
- группы по анализируемому столбцу;
- количество просроченных кредитов по группам;
- количество всех клиентов по группам;
- доля просроченных кредитов по группе от общего числа клиентов в группе (в %).


In [23]:
def category_calculation(row,column):
    data = row.groupby(column).agg({'debt':'sum'})
    data['count'] = row.groupby(column)['debt'].count()
    data['part_debtors'] = data['debt'] / data['count'] * 100
    return data

<a id = 'children'></a>

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Выдвинем гипотезу: дети — это плюс, но большое число иждивенцев может понизить рейтинг. Заемщику будет трудно платить по кредиту.

In [24]:
data_children = category_calculation(data,'children') # вызовим метод category_calculation 
data_children.sort_values(by='part_debtors', ascending=True)

debt  count  part_debtors
children                           
5            0      9      0.000000
0         1063  14091      7.543822
3           27    330      8.181818
1          444   4808      9.234609
2          194   2052      9.454191
4            4     41      9.756098

Данные таблицы показывают, что клиенты без детей более ответственно относятся к выплате кредита в срок. Возможно это связано с тем, что нет расходов на еще одного члена семьи, а значит ежемесячно можно с аккумулировать больше средств, а так же, нет непредвиденных расходов, которые так характерны для семьи с детьми. То есть отсутствие детей не накладывает дополнительные ограничения на семью.

На втором месте расположились клиенты с 3-я детьми, чем серьезно удивили. Предполагаю, что семьи с 3-я детьми оказались более ответственные, чем с 1 или 2 детьми. Кредит им получить тяжелее, чем семье с 1-2 детьми, поэтому нужно поддерживать хорошую кредитную история для будущих кредитов. Так же, по моему мнению, те, кто решился на 3 ребенка более реально оценивают свое финансовое положение.

На 3 ,4 и 5 местах расположились семьи с 1-м, 2-я и 4-я детьми соответственно, они не так сильно придерживаются сроков выплат кредита. 


Рассчитаем долю каждой группы(в %)

In [25]:
total = data_children['count'].sum()
data_children['part_count'] = data_children['count']/total*100
data_children['part_count']

children
0    66.058788
1    22.539965
2     9.619802
3     1.547044
4     0.192209
5     0.042192
Name: part_count, dtype: float64

Самую большую группу обратившихся за кредитом составляю люди без детей (66%). Эта же группа проявила себя как более сознательные в вопросе своевременного возвращения кредита.

Люди, у которых 4 и более детей в семье - это редкость.

Люди с 3 детьми в семье немного «поломали» выдвинутую гипотезу. Но так как их доля составляет всего 1,5%, то предлагаю выделить две категории баллов по надежности возврата кредита:

- `0 детей` – категория балл `1`,

- `есть дети` – категория балл `0`.

Введем в наш датафрейм столбец `balls_children` и повторим расчеты. 

Предварительно обьявим метод, который позволит и следующие признаки(столбцы) приводить к нужным категориям - баллам.
На вход метод принимает датафрейм, в теле метода используем заранее созданный словарь (для каждого столбца - призанака свой) для определения укрупненных категорий - баллов. На выходе получаем балл по каждому элементу столбца-признака.

In [26]:
def balls(row):    
    for key, values in balls_column_dict.items():        
        if key == row:
            return values

Следуя нашей гипотезе зададим систему балов для столбца `children` :

- `0 детей` - категория балл`1`,

- `есть дети` - категория балл `0`.

Для этого создадим словарь балов и применим метод `balls`.

Словарь состоит из:

- ключ - это группа по столбцу - признаку,

- значение соответствующее ключу - это баллы.

In [27]:
balls_column_dict = {0:1, 1:0, 2:0, 3:0, 4:0, 5:0} 
data['balls_children'] = data['children'].apply(balls)
data[['balls_children', 'children']] 
# проверим: добавился ил столбец в датафраме,а так же коррекность заполения нового столбца

balls_children  children
0                   0         1
1                   0         1
2                   1         0
3                   0         3
4                   1         0
...               ...       ...
21520               0         1
21521               1         0
21522               0         1
21523               0         3
21524               0         2

[21331 rows x 2 columns]

Применим метод `category_calculation` к укрупненным группам (`balls_children`) и проверим гипотезу.

In [28]:
data_children_balls = category_calculation(data,'balls_children') # вызовим метод category_calculation
data_children_balls.sort_values(by='part_debtors', ascending=True)

debt  count  part_debtors
balls_children                           
1               1063  14091      7.543822
0                669   7240      9.240331

**Вывод:** 

Гипотезу подтвердили -  наличие детей снижает рейтинг - бал заемщика, так как заемщику труднее платить по кредиту. 

Хотелось бы отметить, что доля не своевременно выплаченных кредитов в двух категориях(по баллам) отличается менее чем на 2%. А это значит, что отсутствие детей если и влияет на своевременность выплат по кредиту, то не радикально.


<a id = 'sample_imbalance'></a>

Данные нашего дата фрейма содержат непропорциональную долю наблюдений в каждой группе (группы по количеству детей в семье). Дисбаланс может привести к появлению ложного впечатления о высокой точности данных анализа, так как входные данные смещены в сторону одной группы, которая показывает лучшие результаты. Выделение двух категорий(балов) позволило немого сбалансировать выборку (добились соотношения 66/44).

<a id = 'family_status'></a>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

Выдвинем гипотезу: люди, "не обремененные" семьей менее ответственные. Ведь взять кредит и вовремя его вернуть гораздо проще, чем прожить всю жизнь с одним человеком!

In [29]:
data_family_status = category_calculation(data,'family_status') # вызовим метод category_calculation 
data_family_status.sort_values(by='part_debtors', ascending=True)

debt  count  part_debtors
family_status                                   
вдовец / вдова           63    951      6.624606
в разводе                84   1189      7.064760
женат / замужем         927  12261      7.560558
гражданский брак        385   4134      9.313014
Не женат / не замужем   273   2796      9.763948

Рассчитаем долю каждой группы(в %)

In [30]:
total = data_family_status['count'].sum()
data_family_status['part_count'] = data_family_status['count']/total*100
data_family_status['part_count']

family_status
Не женат / не замужем    13.107684
в разводе                 5.574047
вдовец / вдова            4.458300
гражданский брак         19.380245
женат / замужем          57.479724
Name: part_count, dtype: float64

Что ж, данные по расчетам в таблице не очень утешительные. Исходя из них, кредиты нужно давать вдовцам, а не семейным или холостым. Нужно отметить, что одинокие люди в выборке составляют 23% (с социальной и демографической точки зрения это не так много и хорошо)

Люди создавшие "ячейку общества" (`женат/замужем` и `гражданский брак`) оказались менее сознательными.

Самой не сознательной категорией выявилась `Не женат/не замужем`.

Не смотря на то, что у нас в датафрейме уже есть идентификатор семейного положения, данные несбалансированы. Предлагаю укрупнить группы и создать новые категории - баллы:

- одинокие ( вдовцы и разведенные) - категория балл`2`,

- состоят в отношениях/браке - категория балл`1`,

- не женат/ не замужем - категория балл`0`

Проверим гипотезу (люди, "не обремененные" семьей, более надежные заемщики)


In [31]:
#создаем словарь категорий
balls_column_dict = {'вдовец / вдова':2, 'в разводе':2, 'женат / замужем':1, 'гражданский брак':1, 'Не женат / не замужем':0} 
data['balls_family_status'] = data['family_status'].apply(balls)
data[['balls_family_status', 'family_status']].tail(10) 
# проверим: добавился ил столбец в датафраме,а так же коррекность заполения нового столбца

balls_family_status     family_status
21515                    1   женат / замужем
21516                    1   женат / замужем
21517                    1  гражданский брак
21518                    1   женат / замужем
21519                    2         в разводе
21520                    1  гражданский брак
21521                    1   женат / замужем
21522                    1  гражданский брак
21523                    1   женат / замужем
21524                    1   женат / замужем

Применим метод category_calculation к укрупненным группам по баллам (balls_family_status) и проверим гипотезу.

In [32]:
data_family_status_balls = category_calculation(data,'balls_family_status') # вызовим метод category_calculation
data_family_status_balls.sort_values(by='part_debtors', ascending=True)

debt  count  part_debtors
balls_family_status                           
2                     147   2140      6.869159
1                    1312  16395      8.002440
0                     273   2796      9.763948

**Вывод:** 

Гипотезу подтвердили частично- люди,ранее состоявшие в браке, но одинокие, более надежные заемщики.

Люди создавшие "ячейку общества" оказались менее ненадежными заемщика, чем ранее сотоявшие в браке и лучше возвращают кредиты, чем холостые.

Хотелось бы отметить, что доля не своевременно выплаченных кредитов в категориях женатых и холостых (по разным причинам) отличается менее чем на 2%. как в худшую, так и лучшую сторону. А это значит, что семья все же влияет на сознательность человека, позволяет подходить более разумно к своим желаниям и возможностям.

<a id = 'total_income'></a>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Выдвинем гипотезу: чем больше мы имеем, тем легче отдаем.

Воспользуемся ранее проведенной категоризацией доходов:
- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.

Для этого используем столбец-признак `total_income_category`.


In [33]:
data_total_income_category  = category_calculation(data,'total_income_category') # вызовим метод category_calculation 
data_total_income_category.sort_values(by='part_debtors', ascending=True)

debt  count  part_debtors
total_income_category                           
D                        21    349      6.017192
B                       354   5014      7.060231
A                         2     25      8.000000
C                      1353  15921      8.498210
E                         2     22      9.090909

Зависимости между уровнем дохода и возвратом кредита в срок не наблюдается.

Рассчитаем долю каждой категории в общей выборке(в %)

In [34]:
total = data_total_income_category['count'].sum()
data_total_income_category['part_count'] = data_total_income_category['count']/total*100
data_total_income_category['part_count']

total_income_category
A     0.117200
B    23.505696
C    74.637851
D     1.636116
E     0.103136
Name: part_count, dtype: float64

Оказывается, что самым часто встречающимся заемщиком являются заемщики группы `С` с доходом `50 001–200 000`, где доход превыщает среднюю заработную плату по стране. Но они хуже всех (не считая группу `Е`) отдаю кредиты.
Менее всего берут кредиты группы `А` и `Е`. Это вполне закономерно, группа `А` может свои потребности удовлетворить за счет доходов (хотя это не помешало им попасть в тройку антирейтинга по возврату кредитов), а группа `Е` не сможет вернуть кредит из-за маленького дохода (что потверждается высоким процентом не своевременного возврата кредита).

Попробуем запустить для исследования другие категории - баллы :

- объедими группы `А`, `В` и `С` в одну категорию - категория балл`0`,

- объедими группы `D` и `Е` в одну категорию - категория балл`1`.

In [35]:
#создаем словарь категорий
balls_column_dict = {'A':0, 'B':0, 'C':0, 'D':1, 'E':1} 
data['balls_total_income'] = data['total_income_category'].apply(balls)
data[['balls_total_income', 'total_income_category']].head(10) 
# проверим: добавился ил столбец в датафраме,а так же коррекность заполения нового столбца

balls_total_income total_income_category
0                   0                     B
1                   0                     C
2                   0                     C
3                   0                     B
4                   0                     C
5                   0                     B
6                   0                     B
7                   0                     C
8                   0                     C
9                   0                     C

In [36]:
data_total_income_balls = category_calculation(data,'balls_total_income') # вызовим метод category_calculation
data_total_income_balls.sort_values(by='part_debtors', ascending=True)

debt  count  part_debtors
balls_total_income                           
1                     23    371      6.199461
0                   1709  20960      8.153626

Как ни странно, но по выборке доходов сложно найти зависимость надежности клиентов от их дохода. В целом, данные показывают, что заемщики с доходом ниже средней заработной платы по стране ( менее 50 тысяч) кредиты возвращают активнее, чем заемщики с более высоким доходом. Но частота их обращение с трудом составит 1,8% от общей выборки.

Так как заемщиков с доходом более 50 тыс превалирующая доля, то их хочется упомянуть отдельно и подробнее:

- доля заемщиков с доходом 50тыс - 200тыс самая внушительная - 75%, а кредиты они возвращают хуже всех (8.50% несвоевременных возвратов),

- заемщики с доходом от 200тыс в этой группе более дисциплинированные (7,06%несвоевременных возвратов) , их доля в общей выборке не велика - 24% (в три раза меньше, чем клиентов с доходом 50-200тыс).

Можно сказать, что гипотеза о том, что чем выше доход, тем надежнее заемщик не подтвердилась. 


<a id = 'equal_intervals'></a>

В данной выборке так же наблюдаем несбалансованность данных по группам, а это в свою очередь снижает достоверность анализа.

Разделим данные признака `total_income` на интервалы равного размера. Для этого воспользуемся методом `.qcut()` и посчитаем количество клиентов, приходящихся на каждую новую группу распределения.

In [37]:
data['quantile_total_income']  = pd.qcut(data['total_income'], q = 4) 
data['quantile_total_income'].value_counts()

(107507.0, 142594.0]     5450
(20666.999, 107507.0]    5333
(195842.0, 2265604.0]    5332
(142594.0, 195842.0]     5216
Name: quantile_total_income, dtype: int64

Разбили на 4 группы:

- от 20,7 тыс до 107,5 тыс
- от 107,5 тыс до 142,6 тыс
- от 142,6 тыс до 195,8 тыс
- от 195,8 тыс до 2 2265,6 тыс

In [38]:
data_total_income_category  = category_calculation(data,'quantile_total_income') # вызовим метод category_calculation 
data_total_income_category.sort_values(by='part_debtors', ascending=True)

debt  count  part_debtors
quantile_total_income                           
(195842.0, 2265604.0]   381   5332      7.145536
(20666.999, 107507.0]   427   5333      8.006750
(142594.0, 195842.0]    444   5216      8.512270
(107507.0, 142594.0]    480   5450      8.807339

Заемщики с доходом от 195,8 тыс и более  более дисциплинированные (7,14% несвоевременных возвратов).

Возьмем для дальнейших расчетов новые интервалы сбалансированной выборки:

- от 20,7 тыс до 107,5 тыс - категория `D`
- от 107,5 тыс до 142,6 тыс - категория `C`
- от 142,6 тыс до 195 8 тыс - категория `B`
- от 195,8 тыс и более - категория `A`

Для этого используем метод `categorize_income` и заменим значения в столбце `purpose_category`

In [39]:
def categorize_income(income):
    try:
        if 0 <= income <= 107507:
            return 'D'
        elif 107507 < income <= 142594:
            return 'C'
        elif 142594 < income <= 195842:
            return 'B'       
        elif income > 195842:            
            return 'A'
    except:
        pass

In [40]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

Создадим словарь категорий для распределения баллов.

In [41]:
#создаем словарь категорий
balls_column_dict = {'D':0, 'C':0, 'B':1, 'A':1} 
data['balls_total_income'] = data['total_income_category'].apply(balls)
#data[['balls_total_income', 'total_income_category']].head(10) 
# проверим: добавился ил столбец в датафраме,а так же коррекность заполения нового столбца

In [42]:
data_total_income_balls = category_calculation(data,'balls_total_income') # вызовим метод category_calculation
data_total_income_balls.sort_values(by='part_debtors', ascending=True)

debt  count  part_debtors
balls_total_income                           
1                    825  10548      7.821388
0                    907  10783      8.411388

**Вывод**

При нормализации данных мы практически подтвердили предыдущие выводы (построенные на ненормализованных данных).

В целом, мы получили крайне не последовательную тенденцию:
- что заемщики с высоким доходом группа `A` (от 195,8 тысяч) кредиты возвращают активнее, чем заемщики с более низким доходом,
- в свою очередь, заемщики группы с самым низким доходом группа `D` (до 107,5 тыс) выплачивают хуже, чем с самым высоким (гр. `A`), но лучше, чем в группе `B` с доходом от 142,5 тыс до 195,8 тыс

Можно сказать, что гипотеза о том, что чем выше доход, тем надежнее заемщик подтвердилась частично.


<a id = 'purpose'></a>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

Определимся с гипотезой: на мой взгляд, чем выше у человека амбиции и желания, тем выше самознание и ответственность.

Воспользуемся ранее проведенной категоризацие целей кредита:

- операции с автомобилем,
- операции с недвижимостью,
- проведение свадьбы,
- получение образования.

Для этого используем столбец-признак `purpose_category`.


In [43]:
data_purpose_category = category_calculation(data,'purpose_category') # вызовим метод category_calculation 
data_purpose_category.sort_values(by='part_debtors', ascending=True)

debt  count  part_debtors
purpose_category                                   
операции с недвижимостью   780  10751      7.255139
проведение свадьбы         183   2313      7.911803
получение образования      369   3988      9.252758
операции с автомобилем     400   4279      9.347978

Покупка недвижимости серьезный шаг, к возврату кредита заемщики подходят так же серьезно.

Рассчитаем долю каждой категории в общей выборке(в %)

In [44]:
total = data_purpose_category['count'].sum()
data_purpose_category['part_count'] = data_purpose_category['count']/total*100
data_purpose_category['part_count']

purpose_category
операции с автомобилем      20.060007
операции с недвижимостью    50.400825
получение образования       18.695795
проведение свадьбы          10.843373
Name: part_count, dtype: float64

Полученные данные показывают, что наше общество заемщиков не сильно подвержено влиянию западно-европейских трендов, говорящих о том, что жить можно и в машине, а на пенсии - в доме престарелых.

Для покупки "своей крыши над головой" обращаются за кредитом 50% заемщиков. Строительство, покупка или ремонт жилья несомненно приводит к улучшению качества жизни. К этому вопросу люди подходят ответсвеннней всего и не хотят долго находиться в долговой зависимости.

Свадьба как оказалось дорогое событие, раз это является причиной одной из категорий займа, но с помощью родственников и друзей удается вовремя вернуть кредит.

Хочется надеяться, что получив образование (17% от выборки), сразу же после обучения обращаются за кредитом для покупки автомобиля (20% от выборки). Но цифры по возврату кредита говорят об ином: хуже всего возвращают кредит те, кто получил его на образование и покупку автомобиля.

Возможно образование забирает много времени и не позволяет получать достаточный и стабильный доход для возврата кредита в срок.

Покупая автомобиль многие не задумываются, что вместе с ним преобретают и источник дополнительных забот и расходов (страховка, заправка, техническое обслуживание, а иногда и ДТП).

Так как банку помимо сроков возврата кредита важна и сумма просроченный кредитов, укрупним категории согласно нашей гипотезы и присвоим баллы:
- крупный займ (операции с недвижимостью) - категория балл `1`,
- средний займ (получение образования, проведение свадьбы, операции с автомобилем) - категория балл `0`.

In [45]:
#создаем словарь категорий
balls_column_dict = {'операции с автомобилем':0, 'операции с недвижимостью':1, 'получение образования':0, 'проведение свадьбы':0} 
data['balls_purpose'] = data['purpose_category'].apply(balls)
data[['balls_purpose', 'purpose_category']].head(10) 

# проверим: добавился ил столбец в датафраме,а так же коррекность заполения нового столбца

balls_purpose          purpose_category
0              1  операции с недвижимостью
1              0    операции с автомобилем
2              1  операции с недвижимостью
3              0     получение образования
4              0        проведение свадьбы
5              1  операции с недвижимостью
6              1  операции с недвижимостью
7              0     получение образования
8              0        проведение свадьбы
9              1  операции с недвижимостью

In [46]:
data_purpose_balls = category_calculation(data,'balls_purpose') # вызовим метод category_calculation 
data_purpose_balls.sort_values(by='part_debtors', ascending=True)


debt  count  part_debtors
balls_purpose                           
1               780  10751      7.255139
0               952  10580      8.998110

**Вывод:** 

Можно сказать, что свою гипотезу то мы подтвердили: чем выше сумма займа, тем отвественне заемщик. Но разрыв по возврату "дорогих" займов и "средних" займов не большой  - 1,7%.
Выдача крупных займов банку более экономически выгодна и судя по нашей таблице это более надежный вид займа. 

Если анализировать вопрос, о том как разные цели кредита влияют на возврат его в срок, можно подвести следующие итоги:
- кредит под операции с недвижимостью является самым надежным, процент не возврата самый низкий - 7,2%, в то время как по другим категориям опредляется в промежутке от 7,9 до 9,3%,
- кредиты под оперции с недвижимостью являются самыми часто запрашиваемыми и составляют 50,4% всей выборки,
- хуже всего возвращают кредиты, полученные под операции с авто (9,3% несвоевременных возвратов),
- так же 2 место в антирейтинге по возврету кредитов занимает группа, где указана цель - на образование.
 

<a id = 'total_balls'></a>

***Дополнение к анализу***

Наш датафрей пополнился столбцами с баллами. Рассчитаем по каждому клиенту общий балл. 

Проверим гипотезу: чем выше общий балл, тем надежнее заемщик.

Создадим новый столбец `total_balls`.

Создадим новый датафрейм с баллами по проанализированным признакам, общим баллом и кредитной историей.

In [47]:
data['total_balls'] = data['balls_children'] + data['balls_family_status'] + data['balls_total_income'] + data['balls_purpose']
data_balls = data[['balls_children', 'balls_family_status','balls_total_income', 'balls_purpose', 'total_balls', 'debt']]
data_balls.head()

balls_children  balls_family_status  balls_total_income  balls_purpose  \
0               0                    1                   1              1   
1               0                    1                   0              0   
2               1                    1                   1              1   
3               0                    1                   1              0   
4               1                    1                   1              0   

   total_balls  debt  
0            3     0  
1            1     0  
2            4     0  
3            2     0  
4            3     0

In [48]:
data['total_balls'].unique() # проверим себя, выведем уникальные значения по столбцу total_balls

array([3, 1, 4, 2, 0, 5])

Максимальное количество баллов, которые мог набрать клиент - `5`, минимальное `0`.

Сформирует таблицу с данными для анализа. 

Вызовем метод `category_calculation`.

In [49]:
data_total_balls_category = category_calculation(data,'total_balls') # вызовим метод category_calculation 
data_total_balls_category.sort_values(by='part_debtors', ascending=True)


debt  count  part_debtors
total_balls                           
5              22    408      5.392157
4             226   3386      6.674542
3             581   7936      7.321069
2             627   7202      8.705915
0              13    130     10.000000
1             263   2269     11.591009

Рассчитаем долю каждой категории (суммы баллов) в общей выборке(в %)

In [50]:
total = data_total_balls_category['count'].sum()
data_total_balls_category['part_count'] = data_total_balls_category['count']/total*100
data_total_balls_category['part_count']

total_balls
0     0.609442
1    10.637101
2    33.763068
3    37.204069
4    15.873611
5     1.912709
Name: part_count, dtype: float64

Гипотеза подтверждается: те клиенты, что набрали максимальное количество баллов (5 баллов) возвращают кредит вовремя с большей вероятностью (5,4%), чем другие группы, но их доля в общей выборке мала (всего 1,9%).

По мере снижения баллов растет размер несвоевременных возвратов кредитов.

Группа заемщиков с 0 и 1 баллами согласно нашей гипотезе показываеют самый высокий процент "ненадежности" - 10,0% и 11,6%.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 

В предобработке данных решили вопрос с пропусками данных в столбцах `days_employed` и `total_income`.
Общая информация о данных таблицы показала, что всего строк в таблице 21525, при этом в столбцах `days_employed` и `total_income` заполнены только 19351 строчка. В каждом из этих столбцов недостает 2174 записи.

Судя по количеству недостающих данных (ровно по 2174 в каждом из столбцов), можно предположить, что отсутствие данных связано с человеческим фактором: 
- или оператор не внес (не запросил) данные о размере дохода и стаже заемщика, 
- или может какое-то отделение банка не собирает эти данные. 

Но данные по столбцу `total_incom` несут очень нужную информацию. Для многих банков ключевыми факторами влияющим на решение о выдаче кридита являются доход и стаж. Думаю такие пропуски от оператов набора и переноса данных недопустимы.

Скорее всего отсутствие данных может произойти по технической причине:
- это сбой работы самой БД или при выгрузке БП; 
- сбой работы приложений, при внесении данных в базу; 
- неудачная миграция данных в процессе переноса данных.

Еще нужно отметить следующее:
- пропуски составляют 10% от выборки (2174/21525) - процент не малый и может повлиять на конечный резальтат анализа,
- в пропусках присутствуют данные о 8% (170/2174) несвоевременно возвращенных кредитов,
- если рассматривать всю выборку, то процент не вовремя возвращенных кредитов составляет тоже 8% (1741/21525).

Можно сделать вывод, что при удалении строк с пропусками, результат анализа не изменился бы.
 

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 

Избавится в выборке от пустых значений можно несколькими способами:

- оставить все как есть,

*С чем столкнемся:*
*Пустые значения будут мешать операциям с ячейками и искажать данные.* 

*Например: для подсчета среднего дохода по определенной группе, нужно общий доход (пустые в общий доход не войдут) разделить на количество заемщиков по группе (а в количество по группе войдут). Таким образом среднее значение будет отличаться (ниже) чем реальное.*

- удалить строки с пустыми значениями,

*Это сделать можно, если нет никаких других подходящих способов по заполнению пустых значений в этих столбцах и удаление такого количества данных не скажется на общий результат.*

*Как ранее указывалось – мы могли бы использовать такой способ, так как на общий результат удаленные строки не особо бы повлияли.*

- заменить значения на расчетные, исходя из других данных выборки.

*Пустые значения можно заменить на среднеарифметическое значение, медиану, моду, максимальное или минимальное и тд. Необходимо определиться, какое из приведенных статистических значений лучше заменит пропуски.* 

*Например :*

-*среднеарифметическое значение чувствительно к выбросам в выборке и не всегда отображает реальность. Если выборка имеет нормальное распределение, то можно использовать среднеарифметическое значение,*

-*медианное значение не чувствительно к выбросам, не искажает картину распределения по пользователям,* 

-*среднее арифметическое может совпадать с медианой только в симметричных выборках, а это на практике встречается редко.*

В замене пропущенных значений столбцов мы использовали медианное значение, так в нашем случае его использование можно считать оптимальным: в результате

-не удаляя строки с пропусками мы сохранили данные столбцов (где данные не отсутствуют),

-не внесли искажения в общую статистику из-за пустых значений в столбцах.

Пример [Рассчет среднеарифметического](#mean)


### Шаг 4: общий вывод.

Целью проведенной работы является исследование о возможном влиянии семейное положения и количества детей клиентов банка на факт погашения кредита в срок. 
Банки для мониторинга надежности рассматривают как социальные, так и экономические аспекты жизни заемщика. Социальное положение заемщика рассматривается системой банковского  скоринга так же с точки зрения финансовой стабильности. Так, студентам, пенсионерам, безработным и семьям с маленькими детьми получить заём сложнее, ведь их материальное положение бывает непредсказуемым.

Банк рассматривает следующие основные показатели по каждому заемщику:

- пол: женщины аккуратнее платят по кредитам, чем мужчины.
- возраст: чем старше заемщик, тем он надежнее, кроме студентов и пенсионеров. Их банки считают более рискованными заемщиками и потому рассматриваются как менее желанные заемщики.
- профессия – некоторые профессии считаются более стабильными
- образование: люди, у которых есть профессиональное образование, обычно имеют стабильный доход.
- стаж: маленький стаж на текущей работе может снизить балл, те большой стаж работы и длительное пребывание на последнем месте подтверждает надежность заемщика.
- доход – чем выше, тем лучше. Банки учитывают не только зарплату, но другие активы: вклады в банках, дивиденды от акций.
- пассивы(эти данные предоставлены не были): наличие недвижимости — всегда преимущество, даже если залог банку не нужен.
- срок, на который оформляется заем (эти данные предоставлены не были) – можно подстраховаться и запросить максимально возможный период, но при этом есть вероятность, что банк заподозрит клиента в неуверенности касательно выплат.

В целях поступившего задания, а именно: подготовить аналитические данные для построения модели кредитного скоринга клиентов, была проанализирована выборка текущих клиентов по предложенным параметрам.

Для достижения цели проекта, решили следующие задачи (этапы разработки):

1.  Оценка входных данных (размер выборки; столбцы, их содержание и типы данных; наличие пустых(незаполненных) элементов таблицы, их количество и причины возникновения.
    Всего в таблице 12 столбцов, встречаются следующие типы данных у столбцов: object, int, float. Строк в таблице - 21525.     Столбцы содержат следующую информацию о клиенте:количество детей в семье, общий трудовой стаж в днях, возраст клиента в годах, уровень образования клиента, семейное положение, пол клиента, тип занятости, имел ли задолженность по возврату кредитов, ежемесячный доход, цель получения кредита. 
    
    
2.	Выявление пропусков и их замена корректными значениями.    
    Количество значений в двух столбцах (стаж и доход) отличается. Это говорит о том, что в данных есть пропущенные значения. В строках, где отсутствуют данные в столбце стаж, отсутствуют данные и по доходу.    
    Вероятнее всего такие пропуски появились по техническим причинам, потому что человвеческий фактор в пропуска таких важных данных недопустим. Является ли такая ошибка случайностю или нет, говорить сложно, все свои соображения о выявленных закономерностях необходимо довести до тех. отдела или разработчиков.
    Пропущенные значения в двух столбцах стаж и доход заменены медианными значениями по типам занятости. Доказали эффективность исползования медианы в таких данных.
    
    
3.	Оценка  и обработка аномалий, определение причин возникновения.
    Анамалии обнаружены в столбеце стаж: 75%  значений -отрицательные (заменили на положительные), для группы по типу занятости - пенсионеры кол-во дней трудового стажа исчисляются в 1000 лет (значения анамальные, но их не трогали, возможно в этом есть какая-то логика, надо спросить у разработчиков).
    Анамальные значения показал столбец дети: 20 детей и -1 ребенок, строки были удалены. Могли строки не удалять,а  земенить 20 на 2 (неверняка подразумевали 2 ребенка) и -1 на 0 детей (подразумевалось отсутсвие детей).
    
    
4.  Замена типов данных.
    Заменили вещественный тип данных по столбцу доход на целочисленный (особая точность нам не нужна, хотя задать разработчикам вопрос - почему такое количество знаков после целой части, имеет смысл.) 
    
    
5.	Выявление неявных дубликатов, приведение данных в один вид и удаление дубликатов. 
    В столбце образование выявлены одинаковые значения(уникальные), записанные в разных регистрах. Привели записи в этом столбце к одному регистку - выявили неявные дубликаты.
    После удалили все дубликаты из таблицы (71 строка).
    
    
6.	Лемматизация целей кредита и выявление категорий.
    По двум столбцам цель кредита и доход данные разбили на категории, при этом устранили (путем замены) повторяющие в разных вариациях данные.
    
    
7.	Выдвинули гипотезы.


8.  Ввели понятие баллы.
    Обычно банки используют одновременно несколько скоринговых систем — для разных категорий заемщиков или видов кредитов. Каждая характеристика заемщика оценивается в баллах. Оценив все параметры, программа высчитывает общий скоринговый балл заемщика. У самых надежных клиентов — самый высокий балл и максимальные шансы получить кредит на выгодных условиях.

    Мы предложили свою бальную систему для скоринга.

    Детальную информацию по результатам бальной системы можно посмотреть в разделе с исследованиями по ссылке [анализ "Баллы"](#total_balls)
    
    
9.	Провели сводный анализа для подтверждение каждой гипотезы. 
    Составлены выводы по каждому вопросу. На этапе разбивки данных на категории были обнаружены явная несбалансированность выборки по рассматриваемым столбцам-признакам. Для устранения или минимизации влияния на результат несбалансированности выборки, попытались укрупнить группы для категориальных столбцов-признаков и применили разбивку на раные группы по численной колонке-признаке. 
    
    В рамках анализа были проверены следующие гипотезы:

    **Есть ли зависимость между количеством детей и возвратом кредита в срок?**

    В рамках данного исследования группы клиентов с 4 и 5 детьми - оказались малочисленными и показали противоречивые данные.
    Категория клиентов без детей оказалась самой многочисленной, а процент задержек по кредиту самый низкий - 7.54%. Самый высокий процент задержек по оплатам оказался у клиентов с двумя детьми и составляет 9.45%.
    
    ! Идеальным клиентом считается: семья без детей. Худшим клиентом является: семья  с 4 и более- детьми. 

    Детальную информацию по данной гипотезе можно посмотреть в разделе с исследованиями по ссылке [анализ "Дети"](#children)

    **Есть ли зависимость между семейным положением и возвратом кредита в срок?**

    В результате проверки гипотезы выявлена следующая закономерность - меньше всего просрочек по кредитам у вдов/вдовцев - 6.62%, а больше всего у холостых клиентов - 9.76%. Самой крупной категорией являются клиенты состоящие в браке - 57,48% от всей выборки, процент просрочек по данной категории составляет 7.56%
    
    ! Идеальным клиентом считается: вдовец. Худшим клиентом является: холостые. 

    Детальную информацию по данной гипотезе можно посмотреть в разделе с исследованиями по ссылке [анализ "Семья"](#family_status)

    **Есть ли зависимость между уровнем дохода и возвратом кредита в срок?**

    Исследование данной гипотезы привели к следующему - меньше всего просрочек по кредитам у клиентов с уровнем дохода от 30 000 до 50 000 руб в месяц - 6.02%, больше всего у клиентов с уровнем дохода от 50 000 до 200 000 руб в месяц - 8.50%, при том что эта категория является наиболее многочисленной в выборке - 74,64%.
    Категории клиентов с низкими значениями доходов (ниже 30 000 руб) или высокими (более 1 000 000 руб) мало представлены в выборке и показывают не самые лучшие результаты (9% и 8% несвоевременного возврата соотвественно).
    
    ! Идеальным клиентом считается: человек c высоким доходом (более 195,8 тыс) . Худшим клиентом является: люди с средним доходом (от 107,5 тыс до 142,5 тыс) 

    Детальную информацию по данной гипотезе можно посмотреть в разделе с исследованиями по ссылке [анализ "Доход"](#total_income)

    **Как разные цели кредита влияют на его возврат в срок?**

    Результаты анализа гипотезы говорят о следующем - меньше всего просрочек по кредитам на недвижимость - 7.26%, больше всего по займам на автомобиль - 9.35%. Самой крупной категорией в выборке по займам являются клиенты по операциям с недвижимостью - 50,40%.
    
    ! Идеальным клиентом считается: человек желающий получить кредит для осуществления операций с недвижимостью . Худшим клиентом является: заемщик с целью купить авто. 

    Детальную информацию по данной гипотезе можно посмотреть в разделе с исследованиями по ссылке [анализ "Цель"](#purpose)
    
    **Общий вывод - семейный статус влияет на вероятность платежей по кредиту в срок - люди, бывший в браке чаще плятят в срок чем те, кто не бывали в браке, причем разведенные и овдовевшие платят в срок чаще, чем люди в браке. Чем больше детей, тем чаще платят не в срок, бездетные реже просрачивают оплату.**
    
    
10.  Сводный портрет "хорошего" и "плохого" заемщика.
     Идеального кандидат на возврат кредита по данным этой выборки без учета возраста и образования: Человек вдовствующий, не имеющий детей, с доходом более 196 тыс, который планирует операции с жилой недвижимостью. Это человек, оставщийся один, без поддержки и детей, обладающий жизненным опытом, четко понимающий, что он может расчитывать только на себя и в срок вернуть кредит.
     Кандидат с наименьшой вероятностью выдачи кредита: Холостой человек, живущий один, с средним доходом от 107 тыс до 143 тыс тысяч, желающий купить машину в кредит.


11. Некоторые дополнения и комментарии. 

    Можно было бы так же рассмотреть влияние показателя возраст на надежность заемщика, он дает вполне логическую привязку к жизненным обстоятельствам и уровню ответственности клиентов.

    Уровень образования мог так же быть интересным параметром для рассмотрения надежности клиентов, банки обращают на этот показатель внимание.

    Анализ совокупности нескольких показателей позволи бы определить более точно:

- какая группа, с точки зрения задолженности порождает наибольшие финансовые проблемы,
- какие показатели имеют большее влияния, а какие меньше влияют,
- какие показатели с совокупности с другими оказывают наибольшее влияние и т.д.


12.	Составили общий вывод.